tasks/windows.json

```diff
@@ -0,0 +1,18 @@
+{
+  "description": "Manage the state of Windows services (without a puppet agent)",
+  "input_method": "powershell",
+  "parameters": {
+    "action": {
+      "description": "The operation (start, stop, restart, status) to perform on the service",
+      "type": "Enum[start, stop, restart, status]"
+    },
+    "starttype": {
```

Comment:
nit: Should this parameter name indicate an action will place.  e.g. `set-starttype`

Reply:
Maybe - will see if there's a naming convention in place elsewhere.

Reply:
I think 'action' is implied, in that this is a task which is implied it changes state.

Reply:
So I had started with jamming startup type into the `action` to experiment with the task interaction, but after playing with that a bit, thought that was a bad fit, and makes some operations that could be one step into two instead. The third commit moved out `starttype`, which is more in sync with how it's modeled on Windows / PowerShell cmdlets / etc... but we also end up with one weird scenario as a direct result, which is why I was leaning away from doing this from the beginning.

Setting an `action` of `status` and then also setting `starttype` to any of the values will modify the startup type, which seems a bit counter-intuitive as a consumer of the task.  BUT, it still might be OK / something that can be documented around / something that could be outright rejected in the task. Could also outright remove `status`, but I think that's something that Discovery is interested in querying.

Interested in hearing what you think @jpogran / @glennsarti?

Reply:
I think it's a separate cmdlet for a reason. Setting start type is a different action that setting service state, and one can operate without the other.

I understand wanting to hit the box the least amount of times you can, but IMHO there isn't sufficient error handling here to prevent a user from setting a state that conflicts with a start type. An error setting state will kill the execution, missing a valid setting for start type. I realize that tasks aren't supposed to be idempotent per se, but by conflating the two an error in one leaves the node in an unknown state. You could spend alot of code handling different situations and returning information based on different outcomes, or you could make two tasks.